src/runtime/controller.test.ts

```diff
@@ -0,0 +1,24 @@
+import { validateHash, ValidationError } from "./controller";
+import { describe, it, expect } from "@jest/globals";
+
+describe("validateHash", () => {
```

Comment:
```suggestion
import { describe, it, expect, jest, beforeEach, afterEach } from "@jest/globals";

describe("validateHash", () => {
  let originalExit: (code?: number) => never;

  beforeEach(() => {
    originalExit = process.exit;
    process.exit = jest.fn() as unknown as (code?: number) => never;
  });

  afterEach(() => {
    process.exit = originalExit;
  });
```

Reply:
thanks for the contribution. The unit test is working properly but it is actually exiting with a code of 1. This will mock the exit and allow the tests to pass